stdlib/src/builtin/object.mojo

```diff
@@ -577,32 +587,47 @@ struct _ObjectImpl(CollectionElement, CollectionElementNew, Stringable):
                 )
                 + "'"
             )
+            return
         if self.is_func():
-            return "Function at address " + hex(int(self.get_as_func().value))
+            writer.write(
+                "Function at address " + hex(int(self.get_as_func().value))
+            )
+            return
         if self.is_list():
-            var res = String("[")
+            writer.write(String("["))
             for j in range(self.get_list_length()):
                 if j != 0:
-                    res += ", "
-                res += str(object(self.get_list_element(j)))
-            res += "]"
-            return res
+                    writer.write(", ")
+                writer.write(str(object(self.get_list_element(j))))
+            writer.write("]")
+            return
 
         var ptr = self.get_obj_attrs_ptr()
-        var res = String("{")
+        writer.write(String("{"))
         var print_sep = False
         for entry in ptr[].impl[].items():
             if print_sep:
-                res += ", "
-            res += (
+                writer.write(", ")
+            writer.write(
                 "'"
                 + str(entry[].key)
                 + "' = "
                 + str(object(entry[].value.copy()))
             )
             print_sep = True
-        res += "}"
-        return res
+        writer.write("}")
+        return
+
+    fn __repr__(self) -> String:
+        """Returns the name (in lowercase) of the specific object type."""
+        return self.__str__()
+
+    fn __str__(self) -> String:
+        """Returns the name (in lowercase) of the specific object type."""
+        var output = String("")
+        var writer = output._unsafe_to_formatter()
+        self.format_to(writer)
+        return output
```

Comment:
**Suggestion** We can avoid the need to touch the internal formatter functions if we change this to:

```mojo
fn __str__(self) -> String:
    return String.format_sequence(self)
```